artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTSessionState.java

```diff
@@ -460,13 +462,23 @@ public int generateMqttId(long messageId, long consumerId) {
          synchronized (dataStoreLock) {
             Integer id = artemisToMqttMessageMap.get(generateKey(messageId, consumerId));
             if (id == null) {
+               int start = currentId;
                do {
-                  if (currentId == MQTTUtil.TWO_BYTE_INT_MAX) {
-                     currentId = 0;
+                  currentId++;
+                  // wrap around to the start if we reach the max
+                  if (currentId > MQTTUtil.TWO_BYTE_INT_MAX) {
```

Comment:
I would get rid of the standalone increment above and just use (++currentId > MQTTUtil.TWO_BYTE_INT_MAX) to make it more clear you are checking that the next value rolled over